app/mutations/addresses/release.rb

```diff
@@ -31,10 +32,18 @@ def execute
         info "Skip gateway address=#{address.id} in pool=#{@pool.id}"
 
       else
-        info "Delete address=#{address.id} in pool=#{@pool.id}"
-
-        address.delete!
+        if ping?(address.address)
+          info "#{self.address} still responds to ping, deleting omitted"
+        else
+          info "Delete address=#{address.id} in pool=#{@pool.id}"
+          address.delete!
+        end
       end
     end
+
+    def ping?(ip_address)
+      icmp = Net::Ping::ICMP.new(ip_address.to_s)
```

Comment:
I would pass an explicit `timeout` parameter to `Net::Ping::ICMP`, because this mutation is expected to timeout in all normal cases.

The default timeout seems to be `5`.